docs/guide/mocking.md

```diff
@@ -0,0 +1,253 @@
+# Mocking
```

Comment:
a lot of the documentation is taken from jest

I've put them in `see also` section, but now sure if that's enough